### PR TITLE
Connect rhf create test page

### PR DESCRIPTION
**Pull request by nikitakukharchuk**
I make a create test that include create quiz and create questions.

### CODE REVIEW

File: src/components/common/Tabs/QuestionTabs/QuestionTabs.tsx [hunk 1]
```diff
@@ -1,40 +1,73 @@
...
+  numberOfQuestions: number;
+  activeTab: any;
 };
-export const QuestionTabs = (props: RectangleProgressBarPropsType) => {
-  const [activeTabId, setActiveTabId] = useState(
-    props.activeQuestionId || props.questionsData[0].questionId
-  );
 
+const getNewQuestion = () => ({
+  id: Math.random(), // ?
```
Comment (gitkoshelew) [inline src/components/common/Tabs/QuestionTabs/QuestionTabs.tsx]: just use counter, maybe in closure. Real id will be generated on BE after

File: src/pages/create-test/FieldsComponents/CheckBoxField.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+          control={
+            <Radio
+              sx={{
+                '&, &.Mui-checked': {
+                  color: 'secondary.main',
+                },
+              }}
+            />
+          }
+          label=''
```
Comment (gitkoshelew) [inline src/pages/create-test/FieldsComponents/CheckBoxField.tsx]: maybe you need label from props?

File: src/pages/create-test/FieldsComponents/CreateAnswer/CreateAnswer.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+              <BasketIcon />
+            </Button>
+          </Box>
+        </Stack>
+      ))}
+      <Box sx={{ flexGrow: 0.1, marginY: '0.5rem' }}>
+        <Button
+          variant='contained'
+          color='primary'
+          onClick={() => append({ name: '', checked: false })}
```
Comment (gitkoshelew) [inline src/pages/create-test/FieldsComponents/CreateAnswer/CreateAnswer.tsx]: transfer to separate handleClick

File: src/pages/create-test/FieldsComponents/CreateTopic/CreateTopic.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+          </Button>
+        )}
+      </Box>
+      {showNewTopicInput && (
+        <>
+          <Stack spacing={3}>
+            <TextField
+              id='new-topic'
+              value={newTopicName}
+              placeholder='Type new topic...'
```
Comment (gitkoshelew) [inline src/pages/create-test/FieldsComponents/CreateTopic/CreateTopic.tsx:96]: add tranlation key for text

File: src/pages/create-test/FieldsComponents/SelectorField/SwitchSelector.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import { useMemo } from 'react';
+import SwitchSelector from 'react-switch-selector';
+import { palette } from '../../../../styles/theme/commonDefaultTheme';
+import { SelectorType } from '../../../../types/SelectorType';
+
+export const SwitchSelectors = ({ onPress, type, value }: SelectorType) => {
+  const SelectorsData = useMemo(
+    () =>
+      // eslint-disable-next-line no-nested-ternary
```
Comment (gitkoshelew) [inline src/pages/create-test/FieldsComponents/SelectorField/SwitchSelector.tsx]: better refactor to if () return, rather then adding eslint-disabe

File: src/pages/create-test/create-question/index.tsx [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+    const correctAnswer = options
+      .filter((obj: { checked: boolean }) => obj.checked)
+      .map((obj: { name: string }) => obj.name);
+    const payload = {
+      ...questionData,
+      content: {
+        options: allOptions,
+        correctAnswer,
+      },
+      options: undefined,
```
Comment (gitkoshelew) [inline src/pages/create-test/create-question/index.tsx]: use null instead of undefined

File: src/pages/create-test/create-quiz/index.tsx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+import { ButtonBackHome } from '../../../components/common/ButtonBackHome';
+import { StylizedPaper } from '../../../components/common/StylizedPaper/StylizedPaper';
+import { Layout } from '../../../components/layout/Layout';
+
+const CreateQuiz = () => {
+  const { handleSubmit, control } = useForm<CreateQuizType>();
+  const router = useRouter();
+  const onSubmitQuiz: SubmitHandler<CreateQuizType> = async (quizData) => {
+    const { numberOfQuestions } = quizData;
+    const responseTopic = await axios
```
Comment (gitkoshelew) [inline src/pages/create-test/create-quiz/index.tsx:13]: this request shouldn't be inside component

**Review by gitkoshelew — CHANGES_REQUESTED** [review]

File: src/components/common/Tabs/QuestionTabs/QuestionTabs.tsx (hunk 1 above)
Comment (nikitakukharchuk) [reply src/components/common/Tabs/QuestionTabs/QuestionTabs.tsx]: fix. i forgot to delete this field, couse i create id in another place

File: src/pages/create-test/FieldsComponents/CheckBoxField.tsx (hunk 2 above)
Comment (nikitakukharchuk) [reply src/pages/create-test/FieldsComponents/CheckBoxField.tsx]: Delete them, couse this is unused component.

File: src/pages/create-test/FieldsComponents/CreateAnswer/CreateAnswer.tsx (hunk 3 above)
Comment (nikitakukharchuk) [reply src/pages/create-test/FieldsComponents/CreateAnswer/CreateAnswer.tsx]: fix.

File: src/pages/create-test/FieldsComponents/CreateTopic/CreateTopic.tsx (hunk 4 above)
Comment (nikitakukharchuk) [reply src/pages/create-test/FieldsComponents/CreateTopic/CreateTopic.tsx:96]: add translation.

File: src/pages/create-test/FieldsComponents/SelectorField/SwitchSelector.tsx (hunk 5 above)
Comment (nikitakukharchuk) [reply src/pages/create-test/FieldsComponents/SelectorField/SwitchSelector.tsx]: refactor this part of code using if()

File: src/pages/create-test/create-question/index.tsx (hunk 6 above)
Comment (nikitakukharchuk) [reply src/pages/create-test/create-question/index.tsx]: fix.

File: src/pages/create-test/create-quiz/index.tsx (hunk 7 above)
Comment (nikitakukharchuk) [reply src/pages/create-test/create-quiz/index.tsx:13]: fix.

**Review by gitkoshelew — APPROVED** [review]

**Review by alexDose — APPROVED** [review]

**Pull request merged**